normandy/recipes/filters.py

```diff
@@ -76,9 +107,9 @@ def to_jexl(self):
 class BucketSampleFilter(BaseFilter):
     """
     Sample a portion of the users by defining a series of buckets, evenly
-    distributing users into those buckets, and then selecting a range of those
-    buckets. This is stable: a given Firefox profile will always match or always
-    not match.
+    distributing users into those buckets, and then selecting a range of
+    those buckets. This is stable: a given set of inputs will always match or
+    always not match.
```

Comment:
I don't know why, but I struggle to parse that sentence; *"a given set of inputs will always match or not match."*
It feels like "duh. of course." 
To me, when you say "this is stable" I imagine it means that it's the opposite of unpredictable. If a certain Firefox Profile runs through this filter a second time, the outcome won't be different.

Reply:
I struggle to explain this actually. It is unpredictable (given the inputs, the only way to know the answer is to do the hashing), but not random (the same inputs always produce the same outputs). This is explained well by the functional concept of "purity", but I don't think that's a good thing to write in this documentation. I'll take another stab at this.